File: src/api/MaterialAdvisor/MaterialAdvisor.Application/Models/Topics/Topic.cs
﻿namespace MaterialAdvisor.Application.Models.Topics;

public class Topic
{
    public Guid Id { get; set; }

    public uint Version { get; set; }

    public string? File { get; set; }

    public DateTime? GeneratedAt { get; set; }

    public IEnumerable<Question> Questions { get; set; } = [];

    public IEnumerable<LanguageText> Name { get; set; } = [];
}

File: src/app/material-advisor/src/app/models/topic/Topic.ts
import { LanguageText } from "@shared/models/LanguageText";
import { GUID } from "@shared/types/GUID";
import { Question } from "./Question";

export interface Topic {
  id: GUID | null;
  version: number;
  name: LanguageText[];
  questions: Question[];
}